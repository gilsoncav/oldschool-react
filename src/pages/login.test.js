import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';

jest.mock('antd', () => ({
  ...jest.requireActual('antd'),
  Alert: () => <>This is an alert!!!</>,
}));

import Login from './login';

const mockStore = configureStore();

describe('Login page component', () => {
  it('renders correctly login not expired', () => {
    const store = mockStore({
      router: { location: { search: '?loginExpired=false' } },
    });
    const wrapper = mount(
      <MemoryRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper.find('Login')).toMatchSnapshot();
  });

  it('renders correctly login expired', () => {
    const store = mockStore({
      router: { location: { search: '?loginExpired=true' } },
    });
    const wrapper = mount(
      <MemoryRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper.find('Login')).toMatchSnapshot();
  });
});
