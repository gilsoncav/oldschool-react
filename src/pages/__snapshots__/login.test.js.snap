// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page component renders correctly login expired 1`] = `
<div
  className="ant-row"
>
  <div
    className="ant-col ant-col-6 ant-col-offset-9"
    style={
      Object {
        "marginTop": "10%",
      }
    }
  >
    This is an alert!!!
    <div
      className="clean-card"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="ant-btn btn-secondary ant-btn-lg"
          onClick={[Function]}
          style={
            Object {
              "float": "left",
              "marginLeft": "5px",
              "marginTop": "5px",
            }
          }
          type="button"
        >
          <i
            aria-label="icon: rollback"
            className="anticon anticon-rollback"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="rollback"
              fill="currentColor"
              focusable="false"
              height="1em"
              key="svg-rollback"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M793 242H366v-74c0-6.7-7.7-10.4-12.9-6.3l-142 112a8 8 0 0 0 0 12.6l142 112c5.2 4.1 12.9.4 12.9-6.3v-74h415v470H175c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h618c35.3 0 64-28.7 64-64V306c0-35.3-28.7-64-64-64z"
                key="svg-rollback-svg-0"
              />
            </svg>
          </i>
          <span
            key=".0"
          >
            Back
          </span>
        </button>
      </a>
      <img
        alt="Logo"
        src="logo.png"
        style={
          Object {
            "float": "right",
            "height": "25px",
            "marginRight": "15px",
            "marginTop": "15px",
          }
        }
      />
      <div
        style={
          Object {
            "paddingBottom": "50px",
            "paddingTop": "50px",
          }
        }
      >
        <form
          className="ant-form ant-form-horizontal login-form"
          id="login-form"
          onSubmit={[Function]}
        >
          <div
            className="ant-row ant-form-item"
          >
            <div
              className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <span
                    className="ant-input-affix-wrapper"
                  >
                    <span
                      className="ant-input-prefix"
                    >
                      <i
                        aria-label="icon: user"
                        className="anticon anticon-user"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="user"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          key="svg-user"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                            key="svg-user-svg-0"
                          />
                        </svg>
                      </i>
                    </span>
                    <input
                      autoComplete="username"
                      className="ant-input"
                      data-__field={
                        Object {
                          "name": "username",
                        }
                      }
                      data-__meta={
                        Object {
                          "initialValue": "",
                          "name": "username",
                          "originalProps": Object {
                            "autoComplete": "username",
                            "placeholder": "Username",
                            "prefix": <Icon
                              type="user"
                            />,
                            "type": "text",
                          },
                          "ref": null,
                          "rules": Array [
                            Object {
                              "message": "This field is required!",
                              "required": true,
                              "type": "string",
                            },
                          ],
                          "trigger": "onChange",
                          "validate": Array [
                            Object {
                              "rules": Array [
                                Object {
                                  "message": "This field is required!",
                                  "required": true,
                                  "type": "string",
                                },
                              ],
                              "trigger": Array [
                                "onChange",
                              ],
                            },
                          ],
                          "valuePropName": "value",
                        }
                      }
                      id="login_form_username"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Username"
                      style={null}
                      type="text"
                      value=""
                    />
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
          >
            <div
              className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <span
                    className="ant-input-affix-wrapper"
                  >
                    <span
                      className="ant-input-prefix"
                    >
                      <i
                        aria-label="icon: lock"
                        className="anticon anticon-lock"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="lock"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          key="svg-lock"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
                            key="svg-lock-svg-0"
                          />
                        </svg>
                      </i>
                    </span>
                    <input
                      autoComplete="current-password"
                      className="ant-input"
                      data-__field={
                        Object {
                          "name": "password",
                        }
                      }
                      data-__meta={
                        Object {
                          "name": "password",
                          "originalProps": Object {
                            "autoComplete": "current-password",
                            "placeholder": "Password",
                            "prefix": <Icon
                              type="lock"
                            />,
                            "type": "password",
                          },
                          "ref": null,
                          "rules": Array [
                            Object {
                              "message": "This field is required!",
                              "required": true,
                              "type": "string",
                            },
                          ],
                          "trigger": "onChange",
                          "validate": Array [
                            Object {
                              "rules": Array [
                                Object {
                                  "message": "This field is required!",
                                  "required": true,
                                  "type": "string",
                                },
                              ],
                              "trigger": Array [
                                "onChange",
                              ],
                            },
                          ],
                          "valuePropName": "value",
                        }
                      }
                      id="login_form_password"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Password"
                      style={null}
                      type="password"
                      value=""
                    />
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
          >
            <div
              className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control has-success"
              >
                <span
                  className="ant-form-item-children"
                >
                  <label
                    className="text-primary ant-checkbox-wrapper ant-checkbox-wrapper-checked"
                  >
                    <span
                      className="ant-checkbox ant-checkbox-checked"
                      style={Object {}}
                    >
                      <input
                        checked={true}
                        className="ant-checkbox-input"
                        data-__field={
                          Object {
                            "name": "remember",
                          }
                        }
                        data-__meta={
                          Object {
                            "initialValue": true,
                            "name": "remember",
                            "originalProps": Object {
                              "children": "Remember me",
                              "className": "text-primary",
                              "indeterminate": false,
                            },
                            "ref": null,
                            "trigger": "onChange",
                            "validate": Array [],
                            "valuePropName": "checked",
                          }
                        }
                        id="login_form_remember"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      Remember me
                    </span>
                  </label>
                  <button
                    className="ant-btn ant-btn-primary"
                    onClick={[Function]}
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                    type="submit"
                  >
                    <span
                      key=".0"
                    >
                      Log in
                    </span>
                  </button>
                  <a
                    href="/register"
                    onClick={[Function]}
                  >
                    Not registered? Click here!
                  </a>
                </span>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Login page component renders correctly login not expired 1`] = `
<div
  className="ant-row"
>
  <div
    className="ant-col ant-col-6 ant-col-offset-9"
    style={
      Object {
        "marginTop": "10%",
      }
    }
  >
    <div
      className="clean-card"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="ant-btn btn-secondary ant-btn-lg"
          onClick={[Function]}
          style={
            Object {
              "float": "left",
              "marginLeft": "5px",
              "marginTop": "5px",
            }
          }
          type="button"
        >
          <i
            aria-label="icon: rollback"
            className="anticon anticon-rollback"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="rollback"
              fill="currentColor"
              focusable="false"
              height="1em"
              key="svg-rollback"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M793 242H366v-74c0-6.7-7.7-10.4-12.9-6.3l-142 112a8 8 0 0 0 0 12.6l142 112c5.2 4.1 12.9.4 12.9-6.3v-74h415v470H175c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h618c35.3 0 64-28.7 64-64V306c0-35.3-28.7-64-64-64z"
                key="svg-rollback-svg-0"
              />
            </svg>
          </i>
          <span
            key=".0"
          >
            Back
          </span>
        </button>
      </a>
      <img
        alt="Logo"
        src="logo.png"
        style={
          Object {
            "float": "right",
            "height": "25px",
            "marginRight": "15px",
            "marginTop": "15px",
          }
        }
      />
      <div
        style={
          Object {
            "paddingBottom": "50px",
            "paddingTop": "50px",
          }
        }
      >
        <form
          className="ant-form ant-form-horizontal login-form"
          id="login-form"
          onSubmit={[Function]}
        >
          <div
            className="ant-row ant-form-item"
          >
            <div
              className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <span
                    className="ant-input-affix-wrapper"
                  >
                    <span
                      className="ant-input-prefix"
                    >
                      <i
                        aria-label="icon: user"
                        className="anticon anticon-user"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="user"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          key="svg-user"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                            key="svg-user-svg-0"
                          />
                        </svg>
                      </i>
                    </span>
                    <input
                      autoComplete="username"
                      className="ant-input"
                      data-__field={
                        Object {
                          "name": "username",
                        }
                      }
                      data-__meta={
                        Object {
                          "initialValue": "",
                          "name": "username",
                          "originalProps": Object {
                            "autoComplete": "username",
                            "placeholder": "Username",
                            "prefix": <Icon
                              type="user"
                            />,
                            "type": "text",
                          },
                          "ref": null,
                          "rules": Array [
                            Object {
                              "message": "This field is required!",
                              "required": true,
                              "type": "string",
                            },
                          ],
                          "trigger": "onChange",
                          "validate": Array [
                            Object {
                              "rules": Array [
                                Object {
                                  "message": "This field is required!",
                                  "required": true,
                                  "type": "string",
                                },
                              ],
                              "trigger": Array [
                                "onChange",
                              ],
                            },
                          ],
                          "valuePropName": "value",
                        }
                      }
                      id="login_form_username"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Username"
                      style={null}
                      type="text"
                      value=""
                    />
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
          >
            <div
              className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <span
                    className="ant-input-affix-wrapper"
                  >
                    <span
                      className="ant-input-prefix"
                    >
                      <i
                        aria-label="icon: lock"
                        className="anticon anticon-lock"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="lock"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          key="svg-lock"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
                            key="svg-lock-svg-0"
                          />
                        </svg>
                      </i>
                    </span>
                    <input
                      autoComplete="current-password"
                      className="ant-input"
                      data-__field={
                        Object {
                          "name": "password",
                        }
                      }
                      data-__meta={
                        Object {
                          "name": "password",
                          "originalProps": Object {
                            "autoComplete": "current-password",
                            "placeholder": "Password",
                            "prefix": <Icon
                              type="lock"
                            />,
                            "type": "password",
                          },
                          "ref": null,
                          "rules": Array [
                            Object {
                              "message": "This field is required!",
                              "required": true,
                              "type": "string",
                            },
                          ],
                          "trigger": "onChange",
                          "validate": Array [
                            Object {
                              "rules": Array [
                                Object {
                                  "message": "This field is required!",
                                  "required": true,
                                  "type": "string",
                                },
                              ],
                              "trigger": Array [
                                "onChange",
                              ],
                            },
                          ],
                          "valuePropName": "value",
                        }
                      }
                      id="login_form_password"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Password"
                      style={null}
                      type="password"
                      value=""
                    />
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
          >
            <div
              className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control has-success"
              >
                <span
                  className="ant-form-item-children"
                >
                  <label
                    className="text-primary ant-checkbox-wrapper ant-checkbox-wrapper-checked"
                  >
                    <span
                      className="ant-checkbox ant-checkbox-checked"
                      style={Object {}}
                    >
                      <input
                        checked={true}
                        className="ant-checkbox-input"
                        data-__field={
                          Object {
                            "name": "remember",
                          }
                        }
                        data-__meta={
                          Object {
                            "initialValue": true,
                            "name": "remember",
                            "originalProps": Object {
                              "children": "Remember me",
                              "className": "text-primary",
                              "indeterminate": false,
                            },
                            "ref": null,
                            "trigger": "onChange",
                            "validate": Array [],
                            "valuePropName": "checked",
                          }
                        }
                        id="login_form_remember"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      Remember me
                    </span>
                  </label>
                  <button
                    className="ant-btn ant-btn-primary"
                    onClick={[Function]}
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                    type="submit"
                  >
                    <span
                      key=".0"
                    >
                      Log in
                    </span>
                  </button>
                  <a
                    href="/register"
                    onClick={[Function]}
                  >
                    Not registered? Click here!
                  </a>
                </span>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
